lib/bigtable/mutation.js

```diff
@@ -0,0 +1,332 @@
+ * @param {object} data - The entity data.
+ * @return {object[]}
+ *
+ * @example
+ * Mutation.encodeSetCell({
+ *   follows: {
+ *     gwashington: 1,
+ *     alincoln: 1
+ *   }
+ * });
+ * // [
+ * //   {
+ * //     setCell: {
+ * //       familyName: 'follows',
+ * //       columnQualifier: 'gwashington', // as buffer
+ * //       timestampMicros: -1, // -1 means to use the server time
+ * //       value: 1 // as buffer
+ * //     }
+ * //   }, {
+ * //     setCell: {
+ * //       familyName: 'follows',
+ * //       columnQualifier: 'alincoln', // as buffer
+ * //       timestampMicros: -1,
+ * //       value: 1 // as buffer
+ * //     }
+ * //   }
+ * // ]
+ */
+Mutation.encodeSetCell = function(data) {
+  var mutations = [];
+
+  Object.keys(data).forEach(function(familyName) {
+    var family = data[familyName];
+
+    Object.keys(family).forEach(function(cellName) {
+      var cell = family[cellName];
+
+      if (!is.object(cell)) {
+        cell = {
+          value: cell
+        };
+      }
+
+      var timestamp = cell.timestamp;
+
+      if (is.date(timestamp)) {
+        timestamp = timestamp.getTime();
+      }
+
+      var setCell = {
+        familyName: familyName,
+        columnQualifier: Mutation.convertToBytes(cellName),
+        timestampMicros: timestamp || -1,
+        value: Mutation.convertToBytes(cell.value)
+      };
+
+      mutations.push({ setCell: setCell });
+    });
+  });
+
+  return mutations;
+};
+
+/**
+ * Formats a `delete` mutation to what the proto service expects. Depending
+ * on what data is supplied to this method, it will return an object that can
+ * will do one of the following:
+ *
+ * * Delete specific cells from a column.
+ * * Delete all cells contained with a specific family.
+ * * Delete all cells from an entire rows.
+ *
+ * @param {object} data - The entry data.
+ * @return {object}
+ *
+ * @example
+ * Mutation.encodeDelete([
+ *   'follows:gwashington'
+ * ]);
+ * // {
+ * //   deleteFromColumn: {
+ * //     familyName: 'follows',
+ * //     columnQualifier: 'gwashington', // as buffer
+ * //     timeRange: null // optional
+ * //   }
+ * // }
+ *
+ * Mutation.encodeDelete([
+ *   'follows'
+ * ]);
+ * // {
+ * //   deleteFromFamily: {
+ * //     familyName: 'follows'
+ * //   }
+ * // }
+ *
+ * Mutation.encodeDelete();
+ * // {
+ * //   deleteFromRow: {}
+ * // }
+ *
+ * //-
+ * // It's also possible to specify a time range when deleting specific columns.
+ * //-
+ * Mutation.encodeDelete([
+ *   {
+ *     column: 'follows:gwashington',
+ *     time: {
+ *       start: new Date('March 21, 2000'),
+ *       end: new Date('March 21, 2001')
+ *     }
+ *   }
+ * ]);
+ */
+Mutation.encodeDelete = function(data) {
+  if (!data) {
+    return [{
+      deleteFromRow: {}
+    }];
+  }
+
+  return arrify(data).map(function(mutation) {
+    if (is.string(mutation)) {
+      mutation = {
+        column: mutation
+      };
+    }
+
+    var column = Mutation.parseColumnName(mutation.column);
```

Comment:
We need a test to confirm this is being called correctly and that we are using the value it returns.